## EDA sanity checking plotting and checking missing values 
- Checking 9/11 

In [1]:
import pandas as pd
import matplotlib.pyplot as plt 
import numpy as np
%matplotlib inline

In [2]:
# df = pd.read_csv("data/silver/1920_2024.csv")
df = pd.read_json("data/silver/1920_2024.json")
# df.Ground = df.Ground.astype("Int64") 
# df.Date = pd.to_datetime(df.Date)

In [7]:
df.Aboard[0]

{'total': 35, 'passengers': 31, 'crew': 4}

In [45]:
# Checking 9/11
df[df.Date == "2001-09-11"]

Decade       Date  Time                   Location           Operator  \
2757    2000 2001-09-11  0945       Arlington, Virginia.  American Airlines   
2758    2000 2001-09-11  0847    New York City, New York  American Airlines   
2759    2000 2001-09-11  0903    New York City, New York   United Air Lines   
2760    2000 2001-09-11  1003  Shanksville, Pennsylvania   United Air Lines   

     Flight_Num                          Route           Ac_Type Registration  \
2757         77  Washington D.C. - Los Angeles  Boeing B-757-223       N644AA   
2758         11           Boston - Los Angeles  Boeing 767-223ER       N334AA   
2759        175           Boston - Los Angeles  Boeing B-767-222       N612UA   
2760         93         Newark - San Francisco  Boeing B-757-222       N591UA   

          Cn_Ln                                       Aboard  \
2757  24602/365   {'total': 64, 'passengers': 58, 'crew': 6}   
2758  22332/169  {'total': 92, 'passengers': 81, 'crew': 11}   
2759   21873/41   {'total': 65, 'passengers': 56, 'crew': 9}   
2760  28142/718   {'total': 44, 'passengers': 37, 'crew': 7}   

                                       Fatalities  Ground  \
2757   {'total': 64, 'passengers': 58, 'crew': 6}     125   
2758  {'total': 92, 'passengers': 81, 'crew': 11}    2750   
2759   {'total': 65, 'passengers': 56, 'crew': 9}    2750   
2760   {'total': 44, 'passengers': 37, 'crew': 7}       0   

                                                Summary  
2757  The aircraft was hijacked after taking off fro...  
2758  The aircraft was hijacked shortly after it lef...  
2759  The aircraft was hijacked shortly after it lef...  
2760  The aircraft was hijacked after taking off fro...

For the two plains in 9/11 the ground was counted twice
- We split the count into two 2750 / 2 

In [46]:
print(df.iloc[2759].Ground / 2)
df.loc[2758, "Ground"] = 1375
df.loc[2759, "Ground"] = 1375

1375.0


Compare Data source with wiki to cross reference dates and fatalities 

In [47]:
df[df.Date == "2001-09-11"]

Decade       Date  Time                   Location           Operator  \
2757    2000 2001-09-11  0945       Arlington, Virginia.  American Airlines   
2758    2000 2001-09-11  0847    New York City, New York  American Airlines   
2759    2000 2001-09-11  0903    New York City, New York   United Air Lines   
2760    2000 2001-09-11  1003  Shanksville, Pennsylvania   United Air Lines   

     Flight_Num                          Route           Ac_Type Registration  \
2757         77  Washington D.C. - Los Angeles  Boeing B-757-223       N644AA   
2758         11           Boston - Los Angeles  Boeing 767-223ER       N334AA   
2759        175           Boston - Los Angeles  Boeing B-767-222       N612UA   
2760         93         Newark - San Francisco  Boeing B-757-222       N591UA   

          Cn_Ln                                       Aboard  \
2757  24602/365   {'total': 64, 'passengers': 58, 'crew': 6}   
2758  22332/169  {'total': 92, 'passengers': 81, 'crew': 11}   
2759   21873/41   {'total': 65, 'passengers': 56, 'crew': 9}   
2760  28142/718   {'total': 44, 'passengers': 37, 'crew': 7}   

                                       Fatalities  Ground  \
2757   {'total': 64, 'passengers': 58, 'crew': 6}     125   
2758  {'total': 92, 'passengers': 81, 'crew': 11}    1375   
2759   {'total': 65, 'passengers': 56, 'crew': 9}    1375   
2760   {'total': 44, 'passengers': 37, 'crew': 7}       0   

                                                Summary  
2757  The aircraft was hijacked after taking off fro...  
2758  The aircraft was hijacked shortly after it lef...  
2759  The aircraft was hijacked shortly after it lef...  
2760  The aircraft was hijacked after taking off fro...

In [48]:
df[df.Date == "1977-03-27"]

Decade       Date  Time                  Location  \
4639    1970 1977-03-27  1707  Tenerife, Canary Islands   

                              Operator Flight_Num  \
4639  Pan American World Airways / KLM  1736/4805   

                                              Route  \
4639  Tenerife - Las Palmas / Tenerife - Las Palmas   

                                   Ac_Type   Registration  \
4639  Boeing B-747-121 / Boeing B-747-206B  N736PA/PH-BUF   

                     Cn_Ln                                         Aboard  \
4639  19643/11 / 20400/157  {'total': 644, 'passengers': 614, 'crew': 30}   

                                         Fatalities  Ground  \
4639  {'total': 583, 'passengers': 560, 'crew': 23}       0   

                                                Summary  
4639  Both aircraft were diverted to Tenerife becaus...

In [49]:
df.Date.value_counts()[df.Date.value_counts() > 1]
# Good to know that we can have more than one crash in a single day

Date
2001-09-11    4
1988-08-31    4
1962-11-23    3
1971-07-30    3
1947-05-29    3
             ..
1985-02-19    2
1969-07-26    2
1988-05-26    2
1969-08-03    2
1968-12-24    2
Name: count, Length: 400, dtype: int64

In [54]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5037 entries, 0 to 5036
Data columns (total 14 columns):
 #   Column        Non-Null Count  Dtype         
---  ------        --------------  -----         
 0   Decade        5037 non-null   int64         
 1   Date          5037 non-null   datetime64[ns]
 2   Time          3529 non-null   object        
 3   Location      5032 non-null   object        
 4   Operator      5026 non-null   object        
 5   Flight_Num    1341 non-null   object        
 6   Route         4273 non-null   object        
 7   Ac_Type       5022 non-null   object        
 8   Registration  4761 non-null   object        
 9   Cn_Ln         4366 non-null   object        
 10  Aboard        5037 non-null   object        
 11  Fatalities    5037 non-null   object        
 12  Ground        4993 non-null   Int64         
 13  Summary       4978 non-null   object        
dtypes: Int64(1), datetime64[ns](1), int64(1), object(11)
memory usage: 556.0+ KB
